status/versions.go

```diff
@@ -259,3 +264,21 @@ func kubernetesVersion() (*ExternalServiceInfo, error) {
 	}
 	return nil, err
 }
+
+// Given a url and a user authentication token,
+// check if the user can access to the specific url.
+func canAccessURL(url, authToken string, auth *config.Auth) (bool, error) {
+	// If either url or authToken is empty, abort check
+	if url == "" || authToken == "" {
+		return false, nil
+	}
+
+	if auth.UseKialiToken {
+		auth.Token = authToken
```

Comment:
we should not mutate `auth` as it's a pointer to the global config. If you just pass `auth` by value instead, it would be fine. Or explicitly make a copy of it before, and use that copy.